pkg/policy/rule.go

```diff
@@ -220,10 +244,31 @@ func (r *rule) resolveL3Policy(ctx *SearchContext, state *traceState, result *L3
 	found := 0
 
 	for _, r := range r.Ingress {
-		found += mergeL3(ctx, "Ingress", r.FromCIDR, &result.Ingress)
+		// TODO (ianvernon): GH-1658
+		var allCIDRs []api.CIDR
+		allCIDRs = append(allCIDRs, r.FromCIDR...)
+
+		allResultantAllowedCIDRs := computeResultantCIDRSet(r.FromCIDRSet)
+		// Transform back to api.CIDR.
+		for _, s := range allResultantAllowedCIDRs {
```

Comment:
Returning `[]api.CIDR` would allow to just `allCIDR = append(allCIDR, computeResultantCIDRSet(r.FromCIDRSet)...)`

Reply:
Fixed.